02-lb-git.md

```diff
@@ -0,0 +1,145 @@
+To ensure that this is the case, run
+```
+$ git config --global user.name "Your Name"
+$ git config --global user.email "Your Name <your.name@cern.ch>"
+```
+and put in your information.
+
+Next, connect to [https://gitlab.cern.ch](https://gitlab.cern.ch) and log in
+with your CERN credentials.
+
+Visit [https://gitlab.cern.ch/profile/keys](https://gitlab.cern.ch/profile/keys) and add an SSH key.
+
+Run this LHCb-specific configuration command:
+```
+git config --global lb-use.protocol ssh
+```
+This makes sure the LHCb commands use the ssh protocol instead of https.
+
+In the Gitlab web interface, create a new project by clicking on the "New project" button.
+Give your project the name "LHCbSK".
+
+We will now set up a local LHCb development area that is connected to the git
+repository you just created.
+In order to create a new dev environment, run
+```
+$ lb-dev --list LHCb
+```
+to see which versions of the LHCb project are available.
+Pick the newest one and run
+```
+$ lb-dev --name LHCbSK LHCb/<version>
+```
+where you need to insert the version you picked earlier.
+This will automatically run `git init` to create a new git repository with some initial files (`Makefile`, `CMakeLists.txt`, etc.).
+In order to connect your repository with the remote one, run
+```
+cd LHCbSK
+git remote add origin ssh://git@gitlab.cern.ch:7999/<username>/LHCbSK.git
+```
+where you need to substitute your username.
+Run
+```
+git push -u origin master
+```
+once to define `master` as the remote branch that we should push to by default.
+
+For the purpose of this tutorial, we've set up an `SKTest` project containing
+the package `TheTestPackage`.
+In order to define `SKTest` as your currently active project, run
+```
+git lb-use SKTest
+```
+This will also fetch the `SKTest` git repository from Gitlab.
+
+You can then run
+```
+git lb-checkout SKTest TheTestPackage
+```
+to check out the files inside the `SKTest` project that belong to the
+`TheTestPackage` package.
+This performs `git checkout` under the hood and commits the files into the
+local (synthetic) branch.
+
+You can now run
+```
+make
+```
+to build the project, make some changes, run `make` again, etc.
+Once you're happy with the changes, you can run
+```
+git add <your-changed-files>
+```
+where you have to specify all files with changes that you want to store,
+and
+```
+git commit
+```
+to store them in a new commit.
+This will open up a text editor that will allow you to type in a commit message.
+
+> ## Committing often {.callout}
+>
+> As new commits are only stored in your local repository, there's no cost to
+> committing often. You should try to make a new commit every time you've made
+> modifications that can be considered a single unit of changes.
+
+Once you want to upload your commits for review, run
+```
+git lb-push SKTest <username>-new-feature
+```
+To upload them to the `<username>-new-feature` branch on the main repository.
```

Comment:
To create a branch called `<username>-new-feature` on the main repository and upload there your commits.